apache/superset
# feat: move ace-editor and mathjs to async modules

### SUMMARY

Follow up on #10831,  move `brace` and `mathjs` to async modules so that the initial page load for <strike>dashboards</strike> most pages can be faster.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

On a dashboard page, `brace` loads only after users add a markdown widget to dashboard:

![brace-async-load](https://user-images.githubusercontent.com/335541/92854867-803fb180-f3a6-11ea-987b-e47b1b40a81f.gif)

Optimized bundle config further helps reduce the overall bundle size:

![new-bundle](https://user-images.githubusercontent.com/335541/92855475-28557a80-f3a7-11ea-9864-ae89cd29139c.png)

![new-load-time](https://user-images.githubusercontent.com/335541/92855489-2e4b5b80-f3a7-11ea-97c6-23a35a4c1bd6.png)

Initial page load time with production assets and without cache was reduced by 30% (9.8s -> 6.9s in Fast 3G network).

On Explore and CRUD page, `brace` and `mathjs` are also loaded on demand. 

### TEST PLAN

- Manual testing + a couple of cypress tests

Expected behaviors:

1. Dashboards should not load `brace.js` and `mathjs` by default.
2. Entering edit mode in dashboards with Markdown widgets should preload `brace.js`, even when users haven't started editing the markdown.
3. Dashboards with Line charts should load `mathjs` automatically.
4. In explore view, `mathjs` will load only for charts that support annotation layers.

### ADDITIONAL INFORMATION
